Google.Api.Gax.Rest.Tests/ClientBuilderBaseTest.GetHttpClientInitializerTest.cs

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Google.Apis.Auth.OAuth2;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using static Google.Api.Gax.Rest.Tests.TestServiceCredentials;
+
+namespace Google.Api.Gax.Rest.Tests
+{
+    public partial class ClientBuilderBaseTest
+    {
+        /// <summary>
+        /// Tests for <see cref="ClientBuilderBase{TClient}.GetHttpClientInitializer"/>.
+        /// Separated into their own class as there are quite a few of them.
+        /// Note that *all* tests require the builder to be configured with a
+        /// single credential source: we never want to fetch the application default credential,
+        /// as that's a singleton.
+        /// </summary>
+        public class GetHttpClientInitializerTest
+        {
+            private static readonly string[] s_scopes = { "scope1", "scope2" };
+
+            [Fact]
+            public async Task JsonCredentials_Valid()
+            {
+                var builder = new FakeBuilder(s_scopes, true)
+                {
+                    JsonCredentials = TestServiceAccountJson
+                };
+                await AssertCredentialAsync(builder, AssertBasicProperties);
+            }
+
+            [Fact]
+            public async Task CredentialsPath_Valid()
+            {
+                var path = Path.GetTempFileName();
+                File.WriteAllText(path, TestServiceAccountJson);
+
+                var builder = new FakeBuilder(s_scopes, true)
+                {
+                    CredentialsPath = path
+                };
+                await AssertCredentialAsync(builder, AssertBasicProperties);
+                File.Delete(path);
```

Comment:
Do you have to delete it if it's a temp file?
I never do that, as I think it deletes on its own when it reaches a threshold.
Fine by me if this is a precaution though.

Reply:
We probably don't *have* to, but we might as well. This whole test is copy/pasted from gRPC and changed just enough to make it pass (with the slightly different API we expose).